internal/services/notification/runevents.go

```diff
@@ -97,6 +116,27 @@ func (n *NotificationService) runEventsHandler(ctx context.Context) error {
 				return errors.WithStack(err)
 			}
 
+			err := n.d.Do(ctx, func(tx *sql.Tx) error {
```

Comment:
Doing this here means that we can lose some run events if the process exits before processing the event.